src/auth_unix.c

```diff
@@ -146,13 +145,10 @@ authunix_ncreate(char *machname, uid_t uid, gid_t gid, int len,
 	return (auth);
 
 cleanup_authunix_create:
-	if (auth)
-		mem_free(auth, sizeof(*auth));
-	if (au) {
-		if (au->au_origcred.oa_base)
-			mem_free(au->au_origcred.oa_base, (size_t) len);
-		mem_free(au, sizeof(*au));
-	}
+	mem_free(auth, sizeof(*auth));
+	if (au->au_origcred.oa_base)
```

Comment:
I don't see how this "if" will ever be true. It is only set after the goto above.